tests/ludwig/utils/test_backward_compatibility.py

```diff
@@ -597,22 +599,56 @@ def test_upgrade_model_progress():
 
     new_model_progress = upgrade_model_progress(old_model_progress)
 
-    for stat in ("improvement", "increase_batch_size", "learning_rate_reduction"):
-        assert f"last_{stat}_epoch" not in new_model_progress
-        assert f"last_{stat}_steps" in new_model_progress
-        assert (
-            new_model_progress[f"last_{stat}_steps"]
-            == old_model_progress[f"last_{stat}_epoch"] * old_model_progress["batch_size"]
-        )
-
-    assert "tune_checkpoint_num" in new_model_progress
-
-    assert "vali_metrics" not in new_model_progress
-    assert "validation_metrics" in new_model_progress
-
-    metric = new_model_progress["validation_metrics"]["combined"]["loss"][0]
-    assert len(metric) == 3
-    assert metric[-1] == 0.59
+    assert new_model_progress == {
+        "batch_size": 64,
+        "best_eval_metric_value": 0.5,
```

Comment:
Do we track what the best eval metric name was anywhere?

Reply:
The best eval metric name comes from the config -- `trainer.validation_field` and `trainer.validation_metric`.

If we can get that information from the config, that would be great since these aren't things that change over the course of a training run. cc: @hungcs

Reply:
Hmm, seems there is a bug with this `validation_metric` field, actually. The docs claim it is set to the feature's default validation metric, but the schema code shows it defaults to LOSS:

https://github.com/ludwig-ai/ludwig/blob/master/ludwig/schema/trainer.py#L155

Can you fix this to use the correct default validation metric as part of this PR?

Reply:
Fixed the documentation in the schema. Changing the default validation metric to be the output feature's default_validation_metric will be addressed in https://github.com/ludwig-ai/ludwig/pull/2820.

Reply:
Resolved.